### PR TITLE
Redesign filter sidebar

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

The filter sidebar can become unwieldy, as all available filters are shown at all times.

#### Relevant ticket(s):

https://mitlibraries.atlassian.net/browse/GDT-127

#### How this addresses that need:

This restyles the filter sidebar as dropdown menus. In so doing, it also changes some behavior in the filter logic:

* Number of results is listed in the filter sidebar header.
* Applied filters are highlighted (previously, the 'applied' filter was added, but we hadn't done anything with it).
* Clicking an applied filter removes that filter (this replaces the previous removal mechanism, which removed all applied filters).

#### Side effects of this change:

* Some tests have been updated to reflect the new logic.
* The sidebar elements have an animation that's tied to height. I've set the height to 200px and added a scroll-behavior rule if the number of filters available exceeds that height. This is somewhat different from how the advanced search sidebar works, so I wanted to draw attention to it unless there are any concerns.
* The JS for the dropdown menus is so minimal that I included it in the view as a `script` tag, but I'm happy to load it via Sprockets if that's preferable.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: app/views/search/_filter.html.erb [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
-  <% if @enhanced_query[category.to_sym].present? %>
-    <div><%= link_to "Show all #{nice_labels[category]&.downcase || category}", results_path(remove_filter(@enhanced_query, category)) %>
-    </div>
-  <% end %>
+  <button class="filter-label <%= 'expanded' if @enhanced_query[category.to_sym].present? %>"
+          onClick="toggleFilter(this)"><%= nice_labels[category] || category %></button>
+  <div class="filter-options">
+    <ul class="category-terms list-unbulleted">
+    <% values.each do |term| %>
+      <li class="term">
```
Comment (JPrevost) [inline app/views/search/_filter.html.erb:9]: I'm not entirely sure why, but the scroll overlaps the content initially in Safari. If I select an element (via tab for example) it reflows the display and looks correct.


Expanded but with nothing selected the scrollbar overlaps the content
Screenshot 2024-02-05 at 2 35 09 PM

Tabbed into an element, the overlap is fixed
Screenshot 2024-02-05 at 3 42 42 PM

File: app/views/search/results.html.erb [hunk 2]
```diff
@@ -64,7 +64,8 @@
     <% if @filters.present? %>
       <aside class="col1q filter-container">
         <div id="filters">
-          <h2>Available filters</h2>
+          <h2 class="hd-3">Filter your results</h2>
+          <h3 class="hd-4"><em><%= @pagination[:hits] %> items</em></h3>
```
Comment (JPrevost) [inline app/views/search/results.html.erb:68]: The hits value is a bit confusing for larger sets as displayed here.

Search for "organic chemistry" and it displays `10000 items`. It should probably display something something more like `10,000+ items`. I think we should expect TIMDEX API/OpenSearch to report "more than 10000 items" as "10000" and adjust the UI to anticipate that.

If you want to handle that in a separate ticket that is fine with me.

**Review by JPrevost — CHANGES_REQUESTED** [review]
If we can address the overlapping content before merge that would be great. If that proves hard to track down, I'd be okay with a ticket to loop back on it.

Otherwise this feels like what was requested to me and works with either mouse or keyboard navigation. 👍🏻

**Comment by jazairi** [comment]
The latest commit addresses some initial feedback on this, both from this review and from Darcy (I shared it with her in a meeting yesterday afternoon). There are also a couple of things that came up while I was doing some testing related to the code review/QA feedback. More details in the commit message.

**Review by JPrevost — APPROVED** [review]

**Pull request merged**